django/db/migrations/optimizer.py

```diff
@@ -113,253 +49,17 @@ def optimize_inner(self, operations, app_label=None):
         for i, operation in enumerate(operations):
             # Compare it to each operation after it
             for j, other in enumerate(operations[i + 1:]):
-                result = self.reduce(operation, other, operations[i + 1:i + j + 1])
-                if result is not None:
+                result = operation.reduce(other, operations[i + 1:i + j + 1], app_label)
+                if isinstance(result, list):
                     # Optimize! Add result, then remaining others, then return
                     new_operations.extend(result)
                     new_operations.extend(operations[i + 1:i + 1 + j])
                     new_operations.extend(operations[i + j + 2:])
                     return new_operations
-                if not self.can_optimize_through(operation, other, app_label):
+                # We can't optimize through other.
+                if result is None:
```

Comment:
`if result is False`

See https://github.com/django/django/pull/5957/files#r49288050